### PR TITLE
Drop legacy dependencies support

### LINKED ISSUE #75: Let drop support for rgeo prior 3.0
@keithdoggett @BuonOmo , 
I think it will be beneficial to drop support for rgeo prior 3.x, that will make maintenance easier for future versions.

We could also drop support for unsupported version of active record.

I have a branch that I can open a PR here.

### PR DESCRIPTION
**Pull request by tagliala**
Hello, please review this approach to build the text matrix

Drop support for:
- Ruby < 3.1
- Rails < 7.0
- RGeo < 3 (close #75)

---

Succesfull run: https://github.com/tagliala/rgeo-activerecord/actions/runs/10560520776

---

- Test against latest stable Ruby versions
- Test against AR 7.1 and 7.2
- Test against `jruby-9.4.8.0` and `jruby-head`
- Mark `edge`/`head` tests as `experimental` to allow failures
- Update checkout actions
- Add explicit permissions for improved security

Ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

### CODE REVIEW

**Review by keithdoggett — APPROVED** [review]
Thanks @tagliala this seems good to me. Only note may be to add AR 7.2 to the matrix as well.

**Comment by tagliala** [comment]
Done

**Comment by BuonOmo** [comment]
I'd also remove unmaintained ruby and activerecord versions (which ultimately might let us keep the original matrix?). WDYT?

**Comment by BuonOmo** [comment]
drop support

**Comment by tagliala** [comment]
> But what do you have in mind more than removing it from the matrix ?

Change this:

https://github.com/rgeo/rgeo-activerecord/blob/06a42e32cb1dc221a5aae4edeb3e1b2d88c70a59/rgeo-activerecord.gemspec#L15

Then I usually search documentation and source code because there may be references, conditional checks, or feature detection related to unsupported rails / ruby

Ref (just the latest thing I've worked on): jamesmartin/inline_svg#164

**Comment by tagliala** [comment]
Since 6.1 is going in EOL at the end of the month, I would go for

- Ruby >= 3.1
- Rails >= 7.0
- RGeo >= 3 (is 1.0 considered legacy?)

File: lib/rgeo/active_record/version.rb [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 module RGeo
   module ActiveRecord
-    VERSION = "7.0.1"
+    VERSION = "8.0.0"
```
Comment (keithdoggett) [inline lib/rgeo/active_record/version.rb:5]: @BuonOmo will we have to synchronize this with the postgis adapter PR that you have up? Do we need a major version bump here?

**Review by keithdoggett — APPROVED** [review]
@tagliala looks good! Thank you. My only question is for @BuonOmo about the version and how to sequence releases.

**Review by BuonOmo — APPROVED** [review]
Version is not that important to me, I'll let you guys decide !

**Comment by tagliala** [comment]
Status changed back to Open. This PR will close #75

**Comment by keithdoggett** [comment]
The only thing with the version I'm thinking that might be breaking is that we're enforcing rgeo 3. I'm not sure about best practices for when there's a big dependency upgrade like that, but since this gem is usually just used downstream of an adapter and not independently installed a major change is ok since the postgis adapter is about to undergo a major change too.

**Comment by keithdoggett** [comment]
Thanks @tagliala

**Comment by tagliala** [comment]
Welcome. I didn't check if there is dead code, now that 2.0 and 1.0 have been dropped. I'm not familiar with RGeo internals

**Pull request merged**